src/wms/src/main/java/org/geoserver/wms/WMS.java

```diff
@@ -1550,7 +1561,27 @@ public Filter getTimeElevationToFilter(List<Object> times, List<Object> elevatio
         return result;
     }
 
-    
+    private List<Object> getNearestMatches(ResourceInfo resourceInfo, DimensionInfo dimension, List<Object> values, String dimensionName) throws IOException {
+        NearestMatchFinder finder = NearestMatchFinder.get(resourceInfo, dimension, dimensionName);
+        List<Object> result = new ArrayList<>();
+        for (Object value : values) {
+            Object nearest = finder.getNearest(value);
```

Comment:
There is no timeout control (rendering timeout), the source list can be very ... very long if the call uses something like T1/T2/smallPeriod.

Reply:
Yep, adding some safeguard here